### DISCUSSION TITLE
CTest options for cross compiling?

### DISCUSSION THREAD

**Discussion by jnz86** [microsoft/vscode-cmake-tools#3470] [Q&A] [answered]
I'm using ARM GNU GCC for compiling my project. Presets for Application and Bootloader with Debug and Release each.

My goal is to add our linux 32bit testing environment - which means different target and compiler options. I would like to have a `arm-none-eabi.cmake` and `CMakeLists.txt` files for my project. Then separately in a `/tests` folder I would like a `x86_64-linux-.cmake` and unique CMakeLists.txt that only describes my tests. It's effectively two separate projects. 

**I'd like** `CMake: Build` to still use the ARM toolchain and the Tests panel to show my 32bit linux tests with all the gutter decorations and functionality that a non-cross compiled project would have.

Is this possible?

**Comment by csorvagep** [comment, marked as answer]
Hi, I missed your question, but I have a similar one: https://github.com/microsoft/vscode-cmake-tools/discussions/3599

For now I solved this by using CmakePresets and a CMake variable to distinguish between the two targets, although it's not easy to switch back and forth. But in my question I mentioned two other possible solution, one with ExternalProject which might cause problems with VS Code, and the other is a multi root workspace, which I don't have any experience with.

**Reply by gcampbell-msft** [reply]
CMakePresets.json is certainly the suggested solution here. There will still be the requirement to switch which configuration you're configuring/building though.

While there is still the requirement for switching between configure presets and build presets selected, this is the suggestion rather than using a multi-root workspace. 

I'm not confident enough to suggest one way or the other for ExternalProject, but there may be discussions on Kitware CMake for ideas here: https://gitlab.kitware.com/cmake
